src/resolve-system-dependencies/resolve-system-dependencies.cc

```diff
@@ -0,0 +1,202 @@
+#include "derivations.hh"
+#include "globals.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include <sys/utsname.h>
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <mach-o/loader.h>
+#include <mach-o/swap.h>
+
+using namespace nix;
+
+typedef std::map<string, std::set<string>> SetMap;
+
+static auto cacheDir = Path{};
+
+Path resolveCacheFile(const Path & lib) {
+    Path lib2 = Path(lib);
+    std::replace(lib2.begin(), lib2.end(), '/', '%');
+    return cacheDir + "/" + lib2;
+}
+
+std::set<string> readCacheFile(const Path & file) {
+    return tokenizeString<set<string>>(readFile(file), "\n");
+}
+
+void writeCacheFile(const Path & file, std::set<string> & deps) {
+    std::ofstream fp;
+    fp.open(file);
+    for (auto & d : deps) {
+        fp << d << "\n";
+    }
+    fp.close();
+}
+
+std::string find_dylib_name(FILE *obj_file, struct load_command cmd) {
+  fpos_t pos;
+  fgetpos(obj_file, &pos);
+  struct dylib_command dylc;
+  dylc.cmd = cmd.cmd;
+  dylc.cmdsize = cmd.cmdsize;
+  fread(&dylc.dylib, sizeof(struct dylib), 1, obj_file);
+
+  char *dylib_name = (char*)calloc(cmd.cmdsize, sizeof(char));
+  fseek(obj_file,
+      // offset is calculated from the beginning of the load command, which is two
+      // uint32_t's backwards
+      dylc.dylib.name.offset - (sizeof(uint32_t) * 2) + pos,
+      SEEK_SET);
+  fread(dylib_name, sizeof(char), cmd.cmdsize, obj_file);
+  fseek(obj_file, pos, SEEK_SET);
+  return std::string(dylib_name);
+}
+
+bool seek_mach64_blob(FILE *obj_file, enum NXByteOrder end) {
+  struct fat_header head;
+  fread(&head, sizeof(struct fat_header), 1, obj_file);
+  swap_fat_header(&head, end);
+  for(uint32_t narches = 0; narches < head.nfat_arch; narches++) {
+    struct fat_arch arch;
+    fread(&arch, sizeof(struct fat_arch), 1, obj_file);
+    swap_fat_arch(&arch, 1, end);
+    if(arch.cputype == CPU_TYPE_X86_64) {
+      fseek(obj_file, arch.offset, SEEK_SET);
+      return true;
+    }
+  }
+  return false;
+}
+
+std::set<std::string> runResolver(const Path & filename) {
```

Comment:
Not sure if there's a rationale behind it, but you seem to alternate between camelCase and underscore_case a fair amount in this file.